upi/azure/02_storage.json

```diff
@@ -1,41 +1,159 @@
+    "galleryName": "[concat('gallery_', replace(parameters('baseName'), '-', '_'))]",
+    "imageName": "[parameters('baseName')]",
+    "imageNameGen2": "[concat(parameters('baseName'), '-gen2')]",
+	"imageVersion": "[take(parameters('imageRelease'), len(parameters('imageRelease'))-6)]"
   },
-  "resources" : [
+  "resources": [
     {
-      "apiVersion" : "2018-06-01",
-      "type": "Microsoft.Compute/images",
-      "name": "[variables('imageName')]",
-      "location" : "[variables('location')]",
-      "properties": {
-        "storageProfile": {
-          "osDisk": {
-            "osType": "Linux",
+      "apiVersion": "2021-10-01",
+      "type": "Microsoft.Compute/galleries",
+      "name": "[variables('galleryName')]",
+      "location": "[variables('location')]",
+      "resources": [
+        {
+          "apiVersion": "2021-10-01",
+          "type": "images",
+          "name": "[variables('imageName')]",
+          "location": "[variables('location')]",
+          "dependsOn": [
+            "[variables('galleryName')]"
+          ],
+          "properties": {
+            "architecture": "[parameters('architecture')]",
+            "hyperVGeneration": "V1",
+            "identifier": {
+              "offer": "rhcos",
+              "publisher": "RedHat",
+              "sku": "basic"
+            },
+            "osState": "Generalized",
+            "osType": "Linux"
+          },
+          "resources": [
+            {
+              "apiVersion": "2021-10-01",
+              "type": "versions",
+              "name": "[variables('imageVersion')]",
+              "location": "[variables('location')]",
+              "dependsOn": [
+                "[variables('imageName')]"
+              ],
+              "properties": {
+                "publishingProfile": {
+                  "storageAccountType": "Standard_LRS",
+                  "targetRegions": [
+                    {
+                      "name": "[variables('location')]",
+                      "regionalReplicaCount": "1"
+                    }
+                  ]
+                },
+                "storageProfile": {
+                  "osDiskImage": {
+                    "source": {
+                      "id": "[parameters('storageAccountId')]",
+                      "uri": "[parameters('vhdBlobURL')]"
+                    }
+                  }
+                }
+              }
+            }
+          ]
+        },
+        {
+          "apiVersion": "2021-10-01",
+          "type": "images",
+          "name": "[variables('imageNameGen2')]",
+          "location": "[variables('location')]",
+          "dependsOn": [
+            "[variables('galleryName')]"
+          ],
+          "properties": {
+            "architecture": "[parameters('architecture')]",
+            "hyperVGeneration": "V2",
+            "identifier": {
+              "offer": "rhcos-gen2",
+              "publisher": "RedHat-gen2",
+              "sku": "gen2"
+            },
```

Comment:
just use the market-place image?

Reply:
marketplace images are not supported for arm64